Formula/z/zlib-ng-compat.rb

```diff
@@ -0,0 +1,42 @@
+class ZlibNgCompat < Formula
+  desc "Zlib replacement with optimizations for next generation systems"
+  homepage "https://github.com/zlib-ng/zlib-ng"
+  url "https://github.com/zlib-ng/zlib-ng/archive/refs/tags/2.2.2.tar.gz"
+  sha256 "fcb41dd59a3f17002aeb1bb21f04696c9b721404890bb945c5ab39d2cb69654c"
+  license "Zlib"
+  head "https://github.com/zlib-ng/zlib-ng.git", branch: "develop"
+
+  livecheck do
+    formula "zlib-ng"
+  end
```

Comment:
Should we add it to [`synced_versions_formulae.json`](https://github.com/Homebrew/homebrew-core/blob/master/synced_versions_formulae.json) (with `zlib-ng`)?